mobius-android/src/test/java/com/spotify/mobius/android/MutableLiveQueueTest.java

```diff
@@ -0,0 +1,170 @@
+    fakeLifecycleOwner2 = new FakeLifecycleOwner();
+    liveObserver = new RecordingObserver<>();
+    pausedObserver = new RecordingObserver<>();
+  }
+
+  @Test
+  public void shouldIgnoreDestroyedLifecycleOwner() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY);
+
+    singleLiveData.setObserver(fakeLifecycleOwner1, liveObserver);
+
+    assertThat(singleLiveData.hasObserver(), equalTo(false));
+  }
+
+  @Test
+  public void shouldSendDataToResumedObserver() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    singleLiveData.setObserver(fakeLifecycleOwner1, liveObserver);
+    singleLiveData.post("one");
+    singleLiveData.post("two");
+
+    assertThat(singleLiveData.hasActiveObserver(), equalTo(true));
+    liveObserver.assertValues("one", "two");
+  }
+
+  @Test
+  public void shouldNotQueueEventsWithNoObserver() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    singleLiveData.post("one");
+    singleLiveData.post("two");
+    singleLiveData.setObserver(fakeLifecycleOwner1, liveObserver, pausedObserver);
+
+    assertThat(liveObserver.valueCount(), equalTo(0));
+    assertThat(pausedObserver.valueCount(), equalTo(0));
+  }
+
+  @Test
+  public void shouldSendQueuedEventsWithValidPausedObserver() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
+
+    singleLiveData.setObserver(fakeLifecycleOwner1, liveObserver, pausedObserver);
+    singleLiveData.post("one");
+    singleLiveData.post("two");
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    assertThat(liveObserver.valueCount(), equalTo(0));
+    pausedObserver.assertValues(queueOf("one", "two"));
+  }
+
+  @Test
+  public void shouldSendLiveAndQueuedEventsWhenRunningAndThenPausedObserver() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    singleLiveData.setObserver(fakeLifecycleOwner1, liveObserver, pausedObserver);
+    singleLiveData.post("one");
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
+    singleLiveData.post("two");
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+
+    liveObserver.assertValues("one");
+    pausedObserver.assertValues(queueOf("two"));
+  }
+
+  @Test
+  public void shouldSendQueuedEffectsIfObserverSwitchedToResumedOneClearing() {
+    fakeLifecycleOwner1.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE);
+
+    singleLiveData.setObserver(fakeLifecycleOwner1, s -> {}, s -> {});
+    singleLiveData.post("one");
+    singleLiveData.post("two");
+    fakeLifecycleOwner2.handleLifecycleEvent(Lifecycle.Event.ON_RESUME);
+    singleLiveData.setObserver(fakeLifecycleOwner2, liveObserver, pausedObserver);
+
+    assertThat(liveObserver.valueCount(), equalTo(0));
+    pausedObserver.assertValues(queueOf("one", "two"));
+  }
+
+  @Test
+  public void shouldSendQueuedEffectsIfObserverSwitchedWithoutClearing() {
```

Comment:
Is 'clearing' correct as a test description here? It looks more like it's about whether the observer is active or not before or after adding?

Reply:
(same for the above test)

Reply:
I didn't read it correctly, it's fine.